CoordImporter/Parser/Parsers.cs

```diff
@@ -1,113 +1,137 @@
-                return new MarkData(markName, mapName, map.TerritoryId, map.RowId, instance, new Vector2(x, y));
-            });
+        return DataManagerManager
+               .GetMapDataByName(mapName)
+               .Match(
+                   Result.Success<MapData, string>,
+                   () => $"Failed to find a map with name: {mapName}"
+               )
+               .Map(map =>
+               {
+                   var markName = groups["mark_name"].Value;
+                   var x = float.Parse(groups["x_coord"].Value, CultureInfo.InvariantCulture);
+                   var y = float.Parse(groups["y_coord"].Value, CultureInfo.InvariantCulture);
+                   var instance = groups["instance"]
+                                  .Value.AsMaybe()
+                                  .Select(instance => instance.Length == 0
+                                                          ? null
+                                                          : instance)
+                                  .Select(instance => (uint?)instanceParser.Invoke(instance!))
+                                  .GetValueOrDefault();
+
+                   return new MarkData(markName, mapName, map.TerritoryId, map.RowId, instance, new Vector2(x, y));
+               });
     }
-}
 
-public interface ITrackerParser
-{
-    public static void LogParse(string tracker, string inputLine, GroupCollection groups)
+    protected static string Dump(GroupCollection groups)
     {
-        Plugin.Logger.Debug(
+        var groupStrings = groups.Values.Select(group => $"({group.Name}:{group.Value})");
+        return string.Join(',', groupStrings);
+    }
+
+    protected void LogParse(string tracker, string inputLine, GroupCollection groups)
+    {
+        Logger.Debug(
             "{0} regex matched for input {1}. Groups are {2}",
             tracker,
             inputLine,
-            groups.Dump()
+            Dump(groups)
         );
     }
 
-    bool CanParseLine(string inputLine);
-    
-    Result<MarkData, string> Parse(string inputLine);
+    public abstract bool CanParseLine(string inputLine);
+    public abstract Result<MarkData, string> Parse(string inputLine);
 }
 
-public partial class SirenParser : ITrackerParser
+public class SirenParser : Parsers
 {
     private static readonly char I1Char = SeIconChar.Instance1.ToIconChar();
-    
+
     // For the format "(Maybe: Storsie) {LinkChar}Labyrinthos{I1Char} ( 17  , 9.6 ) "
-    [GeneratedRegex($@"\((Maybe: )?(?<mark_name>[\w '-]+)\) \ue0bb(?<map_name>[\w '-]+)(?<instance>[\ue0b1-\ue0b9]?)\s+\(\s*(?<x_coord>[0-9\.]+)\s*,\s*(?<y_coord>[0-9\.]+)\s*\)")]
-    private static partial Regex SirenRegex();
 
-    public bool CanParseLine(string inputLine) => SirenRegex().IsMatch(inputLine);
+    protected readonly Regex Regex = new Regex(
+        @"\((Maybe: )?(?<mark_name>[\w '-]+)\) \ue0bb(?<map_name>[\w '-]+)(?<instance>[\ue0b1-\ue0b9]?)\s+\(\s*(?<x_coord>[0-9\.]+)\s*,\s*(?<y_coord>[0-9\.]+)\s*\)",
```

Comment:
since we're moving back to regex constructor, we can also go back to using the `SeIconChar` enum for the special characters, rather than the unicode strings. not a huge benefit, but it's convenient because it's more clear what the character is, because it's a named enum.

Reply:
I wasn't paying attention when I moved away from the attribute-based regex. I've put back the nicer SeIconChar approach you had used before